cli/script/command-executor.ts

```diff
@@ -1149,58 +1149,66 @@ export var release = (command: cli.IReleaseCommand): Promise<void> => {
-    }
+            // Check whether the Cordova or PhoneGap CLIs are
+            // installed, and if not, fail early
+            try {
+                which.sync(cordovaCLI);
+            } catch (e) {
+                try {
+                    cordovaCLI = "phonegap";
+                    which.sync(cordovaCLI);
+                } catch (e) {
+                    throw new Error(`Unable to ${cordovaCommand} project. Please ensure that either the Cordova or PhoneGap CLI is installed.`);
+                }
+            }
 
-    log(chalk.cyan(`Running "${cordovaCLI} ${cordovaCommand}" command:\n`));
-    try {
-        execSync([cordovaCLI, cordovaCommand, platform, "--verbose"].join(" "), { stdio: "inherit" });
-    } catch (error) {
-        throw new Error(`Unable to ${cordovaCommand} project. Please ensure that the CWD represents a Cordova project and that the "${platform}" platform was added by running "${cordovaCLI} platform add ${platform}".`);
-    }
+            log(chalk.cyan(`Running "${cordovaCLI} ${cordovaCommand}" command:\n`));
+            try {
+                execSync([cordovaCLI, cordovaCommand, platform, "--verbose"].join(" "), { stdio: "inherit" });
+            } catch (error) {
+                throw new Error(`Unable to ${cordovaCommand} project. Please ensure that the CWD represents a Cordova project and that the "${platform}" platform was added by running "${cordovaCLI} platform add ${platform}".`);
+            }
 
-    try {
-        var configString: string = fs.readFileSync(path.join(projectRoot, "config.xml"), { encoding: "utf8" });
-    } catch (error) {
-        throw new Error(`Unable to find or read "config.xml" in the CWD. The "release-cordova" command must be executed in a Cordova project folder.`);
-    }
+            try {
+                var configString: string = fs.readFileSync(path.join(projectRoot, "config.xml"), { encoding: "utf8" });
+            } catch (error) {
+                throw new Error(`Unable to find or read "config.xml" in the CWD. The "release-cordova" command must be executed in a Cordova project folder.`);
+            }
 
-    var configPromise: Promise<any> = parseXml(configString);
-    var releaseCommand: cli.IReleaseCommand = <any>command;
+            var configPromise: Promise<any> = parseXml(configString);
 
-    releaseCommand.package = outputFolder;
-    releaseCommand.type = cli.CommandType.release;
+            releaseCommand.package = outputFolder;
+            releaseCommand.type = cli.CommandType.release;
 
-    return configPromise
-        .catch((err: any) => {
-            throw new Error(`Unable to parse "config.xml" in the CWD. Ensure that the contents of "config.xml" is valid XML.`);
+            return configPromise
+                .catch((err: any) => {
+                    throw new Error(`Unable to parse "config.xml" in the CWD. Ensure that the contents of "config.xml" is valid XML.`);
+                });
+        })
+        .catch((err: Error) => {
```

Comment:
You don't need to add this catch here. A `.catch()` basically catches a rejected promise, so that subsequent `.then()` clauses can execute. However, if you `throw` within a promise callback as you are doing here, the exception is caught by the promise library (`q`) and wrapped in a rejected promise again. So, this is effectively a no-op.

Note that this function (`releaseCordova`) returns a promise, which is the final output after each `.then()` executes. It is okay for the returned promise to be rejected, and in fact we recommend that for error handling. Note that rejected promises should end up [here](https://github.com/Microsoft/code-push/blob/master/cli/script/cli.ts#L15), where the error will be logged.